### ISSUE TITLE
Problem with ALTER table during Update

### ISSUE THREAD

**Issue by ghost** [PrestaShop/PrestaShop#21948]
#### Describe the bug

Problem with ALTER table during Update

#### Steps to Reproduce

Make an update 1769 --> 1770

#### Additional information

Linux #799710 SMP Wed Jul 10 08:31:28 UTC 2019 x86_64

Version du logiciel serveur : Apache

Version de PHP : 7.2.34
Limite de mémoire : 2048M
Temps maximal d'exécution : 18000
Taille max. pour envoi de fichiers : 10M
info_outline Informations sur la base de données

Version de MySQL : 10.1.45-MariaDB-0+deb9u1
Serveur MySQL : localhost
Nom MySQL : 176
Utilisateur MySQL : 176
Préfixe des tables : ps_
Moteur MySQL : InnoDB
Connecteur MySQL : DbPDO

```
SQL 1.7.7.0 1071 in ALTER TABLE `ps_admin_filter` CONVERT TO CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci: Specified key was too long; max key length is 767 bytes
SQL 1.7.7.0 1071 in ALTER TABLE `ps_alias` CONVERT TO CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci: Specified key was too long; max key length is 767 bytes
SQL 1.7.7.0 1071 in ALTER TABLE `ps_authorization_role` CONVERT TO CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci: Specified key was too long; max key length is 767 bytes
SQL 1.7.7.0 1071 in ALTER TABLE `ps_module_preference` CONVERT TO CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci: Specified key was too long; max key length is 767 bytes
SQL 1.7.7.0 1071 in ALTER TABLE `ps_smarty_lazy_cache` CONVERT TO CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci: Specified key was too long; max key length is 767 bytes
SQL 1.7.7.0 1071 in ALTER TABLE `ps_tab_module_preference` CONVERT TO CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci: Specified key was too long; max key length is 767 bytes
SQL 1.7.7.0 1071 in ALTER TABLE `ps_admin_filter` CONVERT TO CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci: Specified key was too long; max key length is 767 bytes
SQL 1.7.7.0 1071 in ALTER TABLE `ps_alias` CONVERT TO CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci: Specified key was too long; max key length is 767 bytes
SQL 1.7.7.0 1071 in ALTER TABLE `ps_authorization_role` CONVERT TO CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci: Specified key was too long; max key length is 767 bytes
SQL 1.7.7.0 1071 in ALTER TABLE `ps_module_preference` CONVERT TO CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci: Specified key was too long; max key length is 767 bytes
SQL 1.7.7.0 1071 in ALTER TABLE `ps_smarty_lazy_cache` CONVERT TO CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci: Specified key was too long; max key length is 767 bytes
SQL 1.7.7.0 1071 in ALTER TABLE `ps_tab_module_preference` CONVERT TO CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci: Specified key was too long; max key length is 767 bytes
```

**Comment by okom3pom** [comment]
I need to open an other issue for : Column count doesn't match value count at row 14 ?

**Comment by matks** [comment]
Hi @okom3pom let's explore this topic together 😄 

So if I get correctly
- in PS 1.7.7.0 we converted columns to utf8mb4 character set
- when performing the update on your MySQL server you get error messages
- this is because `max key length` is reached

So it would be because of the size of your database 🤔  ? ping @PierreRambaud 

Can you tell us the size of these tables in your server (count + KB size) ?

**Comment by ghost** [comment]
It's a new install only demo product :) 

i think porblem is mysql version or maria db version, i found a lot of topic with this issue

**Comment by okom3pom** [comment]
@hibatallahAouadni 

The fisrt issue is : 

CONVERT TO CHARACTER SET utf8mb4

Second is : 

Column count doesn't match value count

Third is : 

the var_char size 64

We need all for the change log ?

**Comment by PierreRambaud** [comment]
Hi,
It's not because of the size of the database, but the size of the fields. I just don't understand how it is possible to reach more than 767 in ps_admin_filter for example :thinking: 
Did you add indexes on fields, or did you change something?

**Comment by ghost** [comment]
@PierreRambaud it's a new install without change

**Comment by okom3pom** [comment]
https://cloud.ee.wtf/index.php/s/w7cTJY4j2dzedyp

**Comment by okom3pom** [comment]
At 2:52 you can go to 5:06

**Comment by ghost** [comment]
Here the differences : 

176 :
![structure176](https://user-images.githubusercontent.com/5064590/99503659-c1dc7200-297e-11eb-9d93-a7899c0ea5bd.png)

177 : 
![structure177](https://user-images.githubusercontent.com/5064590/99503664-c30d9f00-297e-11eb-8a8c-d8ed81e34c3a.png)

If i run : 

```ALTER TABLE `ps_admin_filter` CHANGE `filter_id` `filter_id` VARCHAR(191) NOT NULL;```

Before 

```ALTER TABLE `ps_admin_filter` CONVERT TO CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci;```

It's ok, il check all tables with the error and it's the same 255 and 191 

I also found a lot of duplicate querys,  the best is to remove in different PR ? 

Regards

**Comment by ghost** [comment]
Something like this, fixing for me but i don't know if a row is greater than 191 ( **cache_id ?** ) 

```
ALTER TABLE `PREFIX_admin_filter` CHANGE `filter_id` `filter_id` VARCHAR(191) NOT NULL;
ALTER TABLE `PREFIX_alias` CHANGE `alias` `alias` VARCHAR(191) NOT NULL;
ALTER TABLE `PREFIX_authorization_role` CHANGE `slug` `slug` VARCHAR(191) NOT NULL;
ALTER TABLE `PREFIX_module_preference` CHANGE `module` `module` VARCHAR(191) NOT NULL;
ALTER TABLE `PREFIX_smarty_lazy_cache` CHANGE `cache_id` `cache_id` VARCHAR(191) NOT NULL;
ALTER TABLE `PREFIX_hook_tab_module_preference` CHANGE `module` `module` VARCHAR(191) NOT NULL;
```

@PierreRambaud

**Comment by PierreRambaud** [comment]
Yes, there are a lot of duplicates, look like a rebase has failed :(
Thanks for your report. We will fix it today:)

**Comment by PierreRambaud** [comment]
Hey @okom3pom are you able to test this pull request #21975? :)

**Comment by Progi1984** [comment]
Fixed by #21975
